backend/substrapp/ledger_utils.py

```diff
@@ -271,12 +273,20 @@ def _call_ledger(call_type, fcn, args=None, kwargs=None):
         except json.decoder.JSONDecodeError:
             raise LedgerInvalidResponse(response)
 
+        # Set the ledger inside timer as a metric
+        if '__metrics__' in response:
+            metrics = response.pop('__metrics__', None)
```

Comment:
Why do you use `pop` here and not `get` ?

Reply:
Good point, this was done by @thibaultrobert, T think the goal is only to get the metrics and remove them from the response as the information won't be used elsewhere, we can confirm tomorrow with Thibault

Reply:
It's exactly that yes. The `__metrics__` field here is not relevant outside the ledger call that's why I poped it. Is there a problem with this approach ?